astropy/table/pprint.py

```diff
@@ -266,51 +266,27 @@ def _pformat_col(self, col, max_lines=None, show_name=True, show_unit=None,
         else:
             col_width = max(len(x) for x in col_strs) if col_strs else 1
 
-            # Center line content and generate dashed headerline
+            # Center line header content and generate dashed headerline
             for i in outs['i_centers']:
                 col_strs[i] = col_strs[i].center(col_width)
             if outs['i_dashes'] is not None:
                 col_strs[outs['i_dashes']] = '-' * col_width
 
-            # Format columns according to alignment
-            justify_methods = {'<': 'ljust', '^': 'center', '>': 'rjust'}
+            # Format columns according to alignment.  `align` arg has precedent, otherwise
+            # use `col.format` if it is a legal alignment string.  If neither applies
+            # then right justify.
+            justify_methods = {'<': 'ljust', '^': 'center', '>': 'rjust', '0=': 'zfill'}
+            align = align or (col.info.format
+                              if col.info.format in justify_methods
+                              else '>')
+
+            # This can only occur if `align` was explicitly provided.
+            if align not in justify_methods:
```

Comment:
I'll think about it, but part of the problem was that this mini-language was entirely undocumented and unvalidated.  I should have refreshed myself with the original thread, but coming at it fresh I didn't even know what was intended by the code (and it looks fragile, like expected `col.format[1]` to exist, though maybe that can't really be a problem?).  

Overall I was just worried by the interactions between the `_pformat_col_iter` processing and this.